File: app/models/job_application.rb
class JobApplication < ApplicationRecord
  belongs_to :applicant
  belongs_to :job_opening

  enum status: %w[interview_adjustment primary_interview secondary_interview final_interview unofficial_offer joining_the_company rejected declined not_entered],
       _suffix: true

  # validations

  # end for validations

  class << self
    def verify_otp?(phone_number, otp_code)
      phone_number = ::Auths::PhoneNumber.new({ phone_number: phone_number, otp_code: otp_code })
      return unless phone_number.valid?

      ::Auths::PhoneVerification.new(phone_number.formatted_phone_number).verify_otp(otp_code)
      find_by(phone_number: phone_number.formatted_phone_number)
    end
  end
end

File: src/models/jobApplication.ts


export enum StatusEnum {
    INTERVIEW_ADJUSTMENT = "interview_adjustment",
    PRIMARY_INTERVIEW = "primary_interview",
    SECONDARY_INTERVIEW = "secondary_interview",
    FINAL_INTERVIEW = "final_interview",
    UNOFFICIAL_OFFER = "unofficial_offer",
    JOINING_THE_COMPANY = "joining_the_company",
    REJECTED = "rejected",
    DECLINED = "declined",
    NOT_ENTERED = "not_entered"
}

export type JobApplicationModel = {
      id: number;
      created_at: Date;
      updated_at: Date;
      applicant_id: number;
      job_opening_id: number;
      status: `${StatusEnum}`;
      approached_by_scout: boolean;
      user_name_disclosed: boolean;
  
}